### DISCUSSION TITLE
Select instruments for staff visualizer and quantize display note and rest

### DISCUSSION THREAD

**Discussion by tomboul** [cifkao/html-midi-player#78] [Q&A] [answered]
Hello,
I tried other readers midi like midi.js and vexflow for the visualizer, but yours is so so so so very simple to use, bravo!

i try to config a staff visualizer to have only one instrument in the staff, but i dont know how to declare instrument, which is the number to declare ? by Channel ?
it's seems to be the order of samples showing in the console :
here for example, [0] is for flute, [2] for acoustic bass ...

![2024-02-28_19h21_44](https://github.com/cifkao/html-midi-player/assets/5411049/aa56dbc6-6104-4dbf-836b-c666635a5240)

Is it possible to give a minimum time value like an eighth note or a double eighth note? or to specify that is a swing
With a song with "swing" like "All of me" i have some unreadable rest,
![2024-02-28_19h15_59](https://github.com/cifkao/html-midi-player/assets/5411049/49f5f7b3-2b5f-4516-b8cc-0d179f41c1a2)

with vexflow and my code, i obtain this score :
![2024-02-28_19h28_12](https://github.com/cifkao/html-midi-player/assets/5411049/b1f2cf44-a8bf-4da3-a758-93ecd4051ac3)

**Comment by cifkao** [comment, marked as answer]
So if I understand correctly, you would like to have the player play all the instruments in the file, but only display one of them in the visualizer? ~~This is currently not supported, but I think it would be a cool feature! Please feel free to submit a feature request under [issues](https://github.com/cifkao/html-midi-player/issues).~~ See [this answer](https://github.com/cifkao/html-midi-player/discussions/78#discussioncomment-8642210).

Regarding quantization, please see #46.

**Comment by tomboul** [comment]
Sorry for my bad english,
You have good understand

in the documentation, for the object visualizer type staff, you have the property "instruments" and you must give a array of numbers, but it's not indicated what type of number to give

https://magenta.github.io/magenta-js/music/interfaces/_core_visualizer_.staffsvgvisualizerconfig.html

So I gave the number indicating the order of appearance of soundfont loads on the console.log, and it works
Here, on the visualizer type staff, i get only the melody track :
![2024-02-28_19h15_59](https://github.com/cifkao/html-midi-player/assets/5411049/5c4ce376-0370-40ab-b74b-55efaeee9663)


for quantize subject, i continue the discussion on https://github.com/cifkao/html-midi-player/discussions/46

**Reply by cifkao** [reply]
@tomboul Cool, I didn't know about this feature of the staff visualizer!

To see which notes have which instrument number assigned, you can log `player.noteSequence.notes`.
